common/lib/dependabot/clients/bitbucket.rb

```diff
@@ -12,12 +12,28 @@ class Unauthorized < StandardError; end
 
       class Forbidden < StandardError; end
 
+      #######################
+      # Constructor methods #
+      #######################
+
+      def self.for_source(source:, credentials:)
+        credential =
+          credentials.
+          select { |cred| cred["type"] == "git_source" }.
+          find { |cred| cred["host"] == source.hostname }
+
+        new(credentials: credential, source: source)
+      end
+
       ##########
       # Client #
       ##########
 
-      def initialize(credentials:)
+      # FIXME: I don't know if changing the constructor here is safe
+      def initialize(credentials:, source: nil)
```

Comment:
Is this save to add an extra parameter here? (I hadn't done any Ruby before this, so I don't know about this.) It seems to match how the other constructors work.

Reply:
Question: what is using this new `source` field? I skimmed and don't see any references.

Reply:
It should be, since you're providing a default value.

Reply:
I didn't add any references to it since I wasn't sure whether I should prefer a `repo` parameter passed to each method, or to use the instance variable. I went ahead and removed the parameter since I don't think `source` applies to Bitbucket anyway.

(There isn't really an alternative location for Bitbucket. Support for Bitbucket Server is going away in Feb 2024 and the ability to configure custom domain names for Bitbucket Cloud isn't on the Atlassian roadmap to be implemented any time soon, so `bitbucket.org`/`api.bitbucket.org` are the only endpoints that we need to consider—unless someone who can test Bitbucket Server requests the feature.)